src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -4980,9 +4980,19 @@ private void unbootstrap(Runnable onFinish) throws ExecutionException, Interrupt
         logger.debug("waiting for batch log processing.");
         batchlogReplay.get();
 
-        setMode(Mode.LEAVING, "streaming hints to other nodes", true);
+        Future<?> hintsSuccess = ImmediateFuture.success(null);
 
-        Future hintsSuccess = streamHints();
+        if (DatabaseDescriptor.getTransferHintsOnDecommission()) 
+        {
+            setMode(Mode.LEAVING, "streaming hints to other nodes", true);
+            hintsSuccess = streamHints();
+        }
+        else
+        {
+            DatabaseDescriptor.setHintedHandoffEnabled(false);
+            HintsService.instance.pauseDispatch();
+            HintsService.instance.deleteAllHints();
+        }
 
         // wait for the transfer runnables to signal the latch.
         logger.debug("waiting for stream acks.");
```

Comment:
Maybe move this log statement into the block for transferring above?

Reply:
We're still doing the main un-bootstrap, so I think it's fine where it is.